lib/lambda-worker/lambda-worker.ts

```diff
@@ -0,0 +1,199 @@
+
+        // Pass through props from lambda props object
+        // Documented here https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda-nodejs.NodejsFunctionProps.html
+        entry: props.lambdaProps.entry,
+        handler: props.lambdaProps.handler,
+        description: props.lambdaProps.description,
+        environment: props.lambdaProps.environment,
+        reservedConcurrentExecutions:
+          props.lambdaProps.reservedConcurrentExecutions,
+        retryAttempts: props.lambdaProps.retryAttempts,
+        role: props.lambdaProps.role,
+        securityGroup: props.lambdaProps.securityGroup,
+        timeout: lambdaTimeout,
+        vpc: props.lambdaProps.vpc,
+        vpcSubnets: props.lambdaProps.vpcSubnets,
+
+        // Optional props which have been validated or set to a default
+        memorySize: memorySize,
+
+        // Enforce the following properties
+        awsSdkConnectionReuse: true,
+        runtime: lambda.Runtime.NODEJS_14_X,
+      }
+    );
+
+    // Add main queue and DLQ as event sources to the lambda
+    // By default, the main queue is enabled and the DLQ is disabled
+    lambdaWorker.addEventSource(
+      new eventSource.SqsEventSource(lambdaQueue, { enabled: true })
+    );
+    lambdaWorker.addEventSource(
+      new eventSource.SqsEventSource(lambdaDLQ, { enabled: false })
+    );
+
+    // Add alerting
+
+    const alarmAction = new cloudwatchActions.SnsAction(props.alarmTopic);
+
+    // Add an alarm on any messages appearing on the DLQ
+    const approximateNumberOfMessagesVisibleMetric = lambdaDLQ.metric(
+      "ApproximateNumberOfMessagesVisible"
+    );
+    const dlqMessagesVisable = new cloudwatch.Alarm(
+      this,
+      `${props.name}-dlq-messages-visable-alarm`,
+      {
+        alarmName: `${props.name}-dlq-messages-visable-alarm`,
+        alarmDescription: `Alarm when the lambda worker fails to process a message and the message appears on the DLQ`,
+        actionsEnabled: true,
+        metric: approximateNumberOfMessagesVisibleMetric,
+        statistic: "sum",
+        period: cdk.Duration.minutes(1),
+        evaluationPeriods: 1,
+        threshold: 1,
+        comparisonOperator:
+          cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+        // Set treatMissingData to IGNORE
+        // Stops alarms with minimal data having false alarms when they transition to this state
+        treatMissingData: cloudwatch.TreatMissingData.IGNORE,
+      }
+    );
+    dlqMessagesVisable.addAlarmAction(alarmAction);
+    dlqMessagesVisable.addOkAction(alarmAction);
+
+    // Add an alarm for the age of the oldest message on the LambdaWorkers main trigger queue
+    const approximateAgeOfOldestMessageMetric = lambdaDLQ.metric(
+      "ApproximateAgeOfOldestMessage"
+    );
+    const queueMessagesAge = new cloudwatch.Alarm(
+      this,
+      `${props.name}-queue-message-age-alarm`,
+      {
+        alarmName: `${props.name}-queue-message-age-alarm`,
```

Comment:
Before you continue to read this, "it's not you, it's me"...

Do we have an agreed convention for naming these sorts of things? I see that we've used hyphens here extensively. In the other constructs, we've [mostly used capitalisation](https://github.com/talis/talis-cdk-constructs/blob/47f7700aaef93d0bb06583a60378a5c5b45bb0b9/lib/cdn-site-hosting/cdn-site-hosting-construct.ts#L119), but [not consistently](https://github.com/talis/talis-cdk-constructs/blob/main/lib/cdn-site-hosting/cdn-site-hosting-construct.ts#L42)

I'd like to get that decided and written into the wiki as a style matter.

My one observation on the matter is that we have character limits in these places, and I've exceeded them on a few occasions already. That might prefer a `CapitalisationStrategyForDoingThis`, but I've thought about that for precisely as long as I've been typing this messages, and I'll bet you've all done the hard yards for me already.

So, what should be our preferred style?

Reply:
we've also mixed it up in this PR.

That makes me wonder whether our style is both hyphens and capitalisation - if that's the case, we need it documented.